patches/minecraft/net/minecraft/item/ItemArmor.java.patch

```diff
@@ -18,30 +27,94 @@
-+                case GOLD:    return Items.field_151043_k;
-+                case IRON:    return Items.field_151042_j;
-+                case DIAMOND: return Items.field_151045_i;
-+                default:      return customCraftingMaterial;
-+            }
++            ItemStack ret = getRepairItemStack();
++            return ret != null ? ret.func_77973_b() : null;
          }
  
          @SideOnly(Side.CLIENT)
+@@ -246,5 +252,66 @@
+         {
+             return this.field_179243_f;
+         }
++
++        /* ======================================== FORGE START =====================================*/
++        /**
++         * Set the repairing material with a {@code Predicate}. One may use {@code OreNamePredicate} for ores
++         * or {@code ItemPredicate} for items.
++         * 
++         * <p><i>Attention: If the object passed in this method is also an {@link Iterable}, its iterator may be called
++         * to give the actual items.</i>
++         *
++         * @param predicate a predicate of {@link ItemStack} represents repairing material
++         * @return this
++         */
++        public ArmorMaterial setRepairMaterial(com.google.common.base.Predicate<ItemStack> predicate)
++        {
++            if (repairMaterial != null || customCraftingMaterial != null) throw new RuntimeException("Can not change already set repair material");
++            repairMaterial = predicate;
++            return this;
++        }
++
++        /**
++         * Give a {@code Predicate} representation of the repairing material.
++         *
++         * @return a predicate of {@link ItemStack} represents repairing material
++         */
++        public com.google.common.base.Predicate<ItemStack> getRepairMaterial()
++        {
++            com.google.common.base.Predicate<ItemStack> ret = repairMaterial;
++            if (ret != null) return ret;
++            Item i = customCraftingMaterial;
++            if (i != null)
++            {
++                ItemStack item = new ItemStack(i, 1, net.minecraftforge.oredict.OreDictionary.WILDCARD_VALUE);
++                return repairMaterial = new net.minecraftforge.oredict.ItemPredicate(item);
++            }
++            return com.google.common.base.Predicates.alwaysFalse();
++        }
++
++        /**
++         * An {@link ItemStack} version of {@link ArmorMaterial#getRepairItem}.
++         *
++         * @return an item that can be used as repairing material, null if the representation as an {@code ItemStack}
++         * is not available.
++         */
++        public ItemStack getRepairItemStack()
++        {
++            Object p = getRepairMaterial();
++            if (p instanceof Iterable) // use Iterable so we don't check twice
++            {
++                Object obj = com.google.common.collect.Iterables.getFirst((Iterable<?>) p, null);
++                if (obj instanceof ItemStack) return (ItemStack) obj;
```

Comment:
Actually we have no direct calls to this method now. Or if we cache the value here, a ```boolean isCached``` field have to be added together with the ```ItemStack``` field as long as there is ```null``` being a valid output. What if we just leave this part of optimization to modders?